### ISSUE TITLE
Needs help, large byte response in disorder

### ISSUE THREAD

**Issue by bryanzhou** [netty/netty#12337]
I use netty to handle HTTP proxy requests.In recent days I found HTTP response will receive in disorder , and my http client can not decode it , when the size of response is up to 5M  bytes (I'm not sure if this will happen in lower size).

### Expected behavior
Write response bytes into socket with right sequence

### Actual behavior
Wrong sequence get from HTTP client

### Minimal yet complete reproducer code (or URL to code)
This handler is my last handler in outbound pipeline, I tried to split message into pieces or merge into a large one , but didn't work at all. 
```
public class MessageToByteHandler extends MessageToByteEncoder<HttpResponseMessage> {
    @Override
    protected void encode(ChannelHandlerContext channelHandlerContext, HttpResponseMessage httpResponseMessage, ByteBuf out) throws Exception {
        out.writeBytes(httpResponseMessage.getPayload());
    }
}

public class HttpResponseMessage {
    public HttpResponseMessage(byte[] payload) {
        this.payload = payload;
    }
    public byte[] getPayload() {
        return payload;
    }
    public void setPayload(byte[] payload) {
        this.payload = payload;
    }
    byte[] payload;
}

```

### Netty version
4.1.75
### JVM version (e.g. `java -version`)
java version "1.8.0_201"
Java(TM) SE Runtime Environment (build 1.8.0_201-b09)
Java HotSpot(TM) 64-Bit Server VM (build 25.201-b09, mixed mode)
### OS version (e.g. `uname -a`)
Linux k3s.33 3.10.0-514.el7.x86_64 #1 SMP Tue Nov 22 16:42:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

**Comment by hyperxpro** [comment]
Disorder?

**Comment by chrisvest** [comment]
Are you sure your proxy is receiving the bytes in the right order, and on the same channel?

**Comment by bryanzhou** [comment]
> Are you sure your proxy is receiving the bytes in the right order, and on the same channel?

Thx for your reply, I've wrote the bytes received into a log file in MessageToByteHandler.class  and it's in the right order.

**Comment by bryanzhou** [comment]
> Disorder?

Yes,  I use netty to  provide a HTTP server ,this server will handle all HTTP request and redirect to another remote location, like  Nginx to do. MessageToByteHandler.class is the last handler in outbound pipeline, I checked the bytes here and the order is right , but in my client, which send requests to this HTTP server , I got bytes in disorder

**Comment by hyperxpro** [comment]
Can you give a code that can reproduce this?

**Comment by normanmaurer** [comment]
@bryanzhou ping
